### PR TITLE
fix(telegram): improve HTML chunking and preserve word boundaries

### PR DESCRIPTION
**Pull request by badgerbees**
## 📝 Description

**Summary:**
- Preserve whitespace/newline breakpoints when Telegram HTML overflow retries split a formatted chunk.
- Avoid mid-word Telegram chunks from the formatter fallback path.

**Changes:**
- `pkg/channels/telegram/telegram.go`: 
  - Rewrote the dynamic HTML chunking loop to strictly ensure forward progress.
  - Implemented a dual-directional whitespace search for natural word boundaries.
  - Added a fallback for monolithic blocks to ensure the bot never gets stuck.

## 🗣️ Type of Change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 📖 Documentation update
- [ ] ⚡ Code refactoring (no functional changes, no api changes)

## 🤖 AI Code Generation
- [ ] 🤖 Fully AI-generated (100% AI, 0% Human)
- [x] 🛠️ Mostly AI-generated (AI draft, Human verified/modified)
- [ ] 👨‍💻 Mostly Human-written (Human lead, AI assisted or none)

## 🔗 Related Issue
N/A

## 📚 Technical Context (Skip for Docs)
- **Problem Overview:**
  Telegram's `parse_mode=HTML` enforces a strict 4096-character limit. While the initial message might be within our 4000-character split threshold, dense Markdown (links, tags) often expands significantly during HTML conversion. Previously, if this expansion exceeded the limit, the bot could:
  1. Enter an infinite recursion loop if the chunking engine was unable to find a safe split point for long monolithic blocks.
  2. Produce "broken" mid-word chunks that were jarring to read and potentially broke HTML syntax.

- **Solution Implementation:**
  - **Smart Resizing**: Automatically calculates a safe retry length when HTML expansion overflows the Telegram limit.
  - **Word-Boundary Awareness**: Searches for spaces or newlines to ensure messages are split between words rather than in the middle of a link or sentence.
  - **Infinite Loop Prevention**: Guaranteed progression in every split attempt, ensuring the bot never gets stuck on the same chunk.

## 🧪 Test Environment
- **Hardware:** PC
- **OS:** Windows 11
- **Model/Provider:** N/A (Automated Tests)
- **Channels:** Telegram

## 📸 Evidence (Optional)
<details>
<summary>Click to view Logs/Test Results</summary>

**Test Results:**
```bash
ok      github.com/sipeed/picoclaw/pkg/channels/telegram        1.861s
```
All existing Telegram tests pass, and the new chunking logic survives simulated high-expansion scenarios without looping.

</details>

## ☑️ Checklist
- [x] My code/docs follow the style of this project.
- [x] I have performed a self-review of my own changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by alexhoshina** [comment]
`make lint` plz

**Comment by badgerbees** [comment]
@alexhoshina fixed it, sorry what a stupid mistake lol. Could you approve workflow again perhaps?

File: pkg/channels/telegram/telegram.go [hunk 1]
```diff
@@ -190,15 +190,85 @@ func (c *TelegramChannel) Send(ctx context.Context, msg bus.OutboundMessage) err
...
+				splitIdx = smallerLen
+			}
+
+			// Ensure we split at a valid index to guarantee forward progress.
+			if splitIdx <= 0 {
+				splitIdx = 1
+			}
+
+			// Attempt to split using the determined index, ensuring code block integrity is maintained.
+			subChunks := channels.SplitMessage(chunk, splitIdx)
```
Comment (alexhoshina) [inline pkg/channels/telegram/telegram.go]: In Send, splitIdx is a character position, but channels.SplitMessage interprets its second argument as the maximum chunk length. If the last whitespace before smallerLen is far back near the start of the chunk (for example "hello " + strings.Repeat("**a**", 700) or a long code block after "Here:\n"), this call becomes SplitMessage(chunk, 5)-style input and shreds one 3–4k reply into hundreds or thousands of 2–3 rune Telegram

Comment (alexhoshina) [reply pkg/channels/telegram/telegram.go]: This is the function signature of SplitMessage  
```go
func SplitMessage(content string, maxLen int) []string {}
```

Comment (badgerbees) [reply pkg/channels/telegram/telegram.go]: I've refactored this to remove the manual index-hunting loop entirely. The code now passes the estimated safe Markdown budget (`smallerLen`) directly to `channels.SplitMessage`. This lets the utility find the best natural break points (spaces/newlines) while keeping chunks as large as possible and preserving code block integrity.

**Comment by alexhoshina** [comment]
Thank you for your contribution!

**Comment by Orgmar** [comment]
@badgerbees Solid fix on the Telegram HTML chunking. The dual-directional whitespace search for word boundaries is a smart approach, and the guaranteed forward progress safeguard should put an end to those infinite loop edge cases with dense markdown. Good to see it backed by tests too.

We have a **PicoClaw Dev Group** on Discord where contributors connect and collaborate. If you'd like to join, send an email to `support@sipeed.com` with subject `[Join PicoClaw Dev Group] badgerbees` and we'll get you the invite link!
